css/selectors/state.json

```diff
@@ -0,0 +1,49 @@
+{
+  "css": {
+    "selectors": {
+      "state": {
+        "__compat": {
+          "description": "Custom state pseudo-class selector (<code>:state()</code>)",
+          "mdn_url": "https://developer.mozilla.org/docs/Web/CSS/:state",
+          "spec_url": "https://html.spec.whatwg.org/multipage/semantics-other.html#selector-custom",
+          "support": {
+            "chrome": {
+              "version_added": false,
+              "notes": "From version 90 Chromium supports custom state selection using a dashed-ident (such as <code>:--foo</code>) instead of <code>:state()</code>."
```

Comment:
Note that Chrome supports an older format from Chrome 90 (assuming support in same versions as https://developer.mozilla.org/en-US/docs/Web/API/CustomStateSet#browser_compatibility - which is logical). It does not yet support `state()` but intends to support it and deprecate the old dashed-ident format - see https://chromestatus.com/feature/5586433790443520

I am choosing to make this the main doc for the concept since the dashed ident is intended to go away eventually. Hence adding supported as `false` and just recording the information about the old support.